docker/Dockerfile

```diff
@@ -16,19 +16,15 @@ RUN cd /usr/src/node-red/nodes/core; \
     awk '{system("mv "$1" "$1".ign")}';
 
 # Add node-red user so we aren't running as root.
+# And add all dojot-related nodes
+COPY contrib-nodes /data/node_modules
 RUN useradd --home-dir /usr/src/node-red --no-create-home node-red \
     && chown -R node-red:node-red /data \
-    && chown -R node-red:node-red /usr/src/node-red
-
-USER node-red
-RUN npm install && npm run build
+    && chown -R node-red:node-red /usr/src/node-red \
+    && chown -R node-red:node-red /data/node_modules
```

Comment:
This last line actually is a no-op: /data/node_modues is already owned by node-red:node-red due to line 20/22